rfc-126-custom-configuration-for-dependabot.md

```diff
@@ -0,0 +1,254 @@
+Top 5 quickest libraries to merge: govuk_publishing_components (0:42:20), sprockets (0:38:34), rack-protection (0:19:10), actionview (0:05:42), webrick (0:01:17)
+
+Ignored Libraries
+=================
+Total PRs: 7378
+Mean time to merge: 3 days, 15:19:59.342911
+Max time to merge: 146 days, 3:40:01
+Top 5 longest libraries to merge: acts-as-taggable-on (146 days, 3:40:01), paper_trail (85 days, 7:45:33), rails-controller-testing (84 days, 3:29:26), mini_magick (41 days, 4:15:20), excon (41 days, 0:57:48)
+Top 5 quickest libraries to merge: docker-compose (0:12:51), vcr (0:10:44), miller-columns-element (0:05:36), faraday_middleware (0:03:30), nltk (0:01:14)
+```
+</details>
+
+## Proposal
+
+We will configure Dependabot to only raise PRs against libraries that fit within one of three categories. Any updates to libraries which don't fit into one of these categories will be ignored and PRs won't be raised for them.
+
+The three categories are:
+
+- [Security updates]
+- Internal libraries - these are libraries written by us (for example `gds-api-adapters`, `govuk_publishing_components`)
+- Framework libraries - these are libraries which our apps heavily rely on to work (for example `rails`, `rspec-rails`)
+
+These categories have been chosen to ensure we stay up to date on libraries which are important to us while also reducing the number of unnecessary version bumps.
+
+[Security updates]: https://help.github.com/en/github/managing-security-vulnerabilities/configuring-github-dependabot-security-updates#about-github-dependabot-security-updates
+
+### Custom configuration
+
+Dependabot [supports configuration files][dependabot-config] stored in each repo to customise its behaviour. This RFC proposes using a configuration file like the one detailed below in each of our repos.
+
+We've already been trialing this out in [Content Publisher] and [govuk_publishing_components], based on some initial experiements in [Signon].
+
+[dependabot-config]: https://dependabot.com/docs/config-file/
+[Content Publisher]: https://github.com/alphagov/content-publisher/pull/2062
+[govuk_publishing_components]: https://github.com/alphagov/govuk_publishing_components/pull/1564
+[Signon]: https://github.com/alphagov/signon/pull/1426
+
+```yaml
+version: 1
+update_configs:
+  - package_manager: ruby:bundler
+    directory: /
+    update_schedule: daily
+    allowed_updates:
+      # Security updates
+      - match: { update_type: security }
+      - match: { dependency_name: brakeman }
+      # Internal gems
+      - match: { dependency_name: gds-api-adapters }
+      - match: { dependency_name: gds-sso }
+      - match: { dependency_name: govspeak }
+      - match: { dependency_name: govuk* }
+      - match: { dependency_name: rubocop-govuk }
+      - match: { dependency_name: plek }
+      - match: { dependency_name: scss_lint-govuk }
+      # Framework gems
+      - match: { dependency_name: factory_bot_rails }
```

Comment:
```suggestion
      - match: { dependency_name: factory_bot }
```

Arguably most of our code relies on the core dependency, rather than the smaller Rails-specific layer on top of it.

Reply:
Sure, added in 6af5beaa9ec1436eba5fda5d37bf15804d0ecac8.